### PR TITLE
General cleanup

### PR DESCRIPTION
**Pull request by nschcolnicov**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/smrnaseq/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/smrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/prepare_genome/main.nf [hunk 1]
```diff
@@ -96,18 +96,18 @@ workflow PREPARE_GENOME {
     emit:
     fasta                 = ch_fasta               // channel: [ val(meta), path(fasta) ]
     has_fasta             = bool_has_fasta         // boolean
-    bowtie_index          = ch_bowtie_index        // channel: [genome.1.ebwt, genome.2.ebwt, genome.3.ebwt, genome.4.ebwt, genome.rev.1.ebwt, genome.rev.2.ebwt]
+    bowtie_index          = ch_bowtie_index        // channel: [ val(meta), path(index) ]
```
[the remark this answers is not in the document]
Comment (nschcolnicov) [reply subworkflows/local/prepare_genome/main.nf:99]: This was updated in a previous PR, it now has a meta , but I forgot to update the channel comment

[the remark this answers is not in the document]
Comment (atrigila) [reply subworkflows/local/prepare_genome/main.nf:99]: Cool. We can simply have the channel to be `[id:"bowtie_index", [workspace/etc/bowtie_index]]`. It does not have to have the path to the `ebwt` files as the modules can handle the whole directory (and `mirdeep` cannot handle it without the directory).

File: modules.json
[the remark this answers is not in the document]
Comment (nschcolnicov) [reply modules.json:1]: This is not straightforward, lets do it on a separate PR: https://github.com/orgs/nf-core/projects/74/views/7?pane=issue&itemId=81583371

**Comment by atrigila** [comment]
Are we addressing the whole scope of #400 or some sub-tasks here (in this PR)? 

- [ ] Address the TODOs left in the pipeline.
- [ ] Check variables, add "ch_" to channel and "val_" to value variables
- [ ] Make sure that variable names remain consistent across workflows.
- [ ] Assess which emitted files from subworkflows are actually being used.
- [ ] Replace any dsl1 or groovy funcitons which can be replaced with nextflow DSL2 functions.

File: workflows/smrnaseq.nf [hunk 2]
```diff
@@ -53,9 +53,9 @@ workflow NFCORE_SMRNASEQ {
     ch_reference_hairpin   // channel: [ val(meta), path(fasta) ]
     ch_mirna_gtf           // channel: [ path(GTF) ]
```
[the remark this answers is not in the document]
Comment (nschcolnicov) [reply workflows/smrnaseq.nf:54]: We don't need it, but will add it for consistency

**Comment by nschcolnicov** [comment]
> Are we addressing the whole scope of #400 or some sub-tasks here (in this PR)?
> 
> * [x]  Address the TODOs left in the pipeline.
> * [x]  Check variables, add "ch_" to channel and "val_" to value variables
> * [x]  Make sure that variable names remain consistent across workflows.
> * [x]  Assess which emitted files from subworkflows are actually being used.
> * [x]  Replace any dsl1 or groovy funcitons which can be replaced with nextflow DSL2 functions.
> * [x]  Reviewing parameters modules and subworkflows to check there are no unused items.

Ah, I see, yes we adressing all of the items in the issue, we could also add, I will also add, reviewing parameters modules and subworkflows to check there are no unused items.

File: subworkflows/local/genome_quant.nf [hunk 3]
```diff
@@ -7,7 +7,7 @@ include { BOWTIE_ALIGN as BOWTIE_MAP_GENOME } from '../../modules/nf-core/bowtie
 
 workflow GENOME_QUANT {
     take:
-    ch_bowtie_index // channel: [genome.1.ebwt, genome.2.ebwt, genome.3.ebwt, genome.4.ebwt, genome.rev.1.ebwt, genome.rev.2.ebwt]
+    ch_bowtie_index // channel: [ val(meta), [ path(genome.1.ebwt), path(genome.2.ebwt), path(genome.3.ebwt), path(genome.4.ebwt), path(genome.rev.1.ebwt), path(genome.rev.2.ebwt) ] ]
```
Comment (atrigila) [inline subworkflows/local/genome_quant.nf:10]: Sorry if I wasn't clear in my previous comment, but for this to work in `mirdeep2`  `ch_bowtie_index` should be of structure:

`channel: [ val(meta), [ path(one_directory)]` and not 
`channel: [ val(meta), [ path(genome.1.ebwt), path(genome.2.ebwt), path(genome.3.ebwt), path(genome.4.ebwt), path(genome.rev.1.ebwt), path(genome.rev.2.ebwt) ] ]`. 

Please write me if we need to review this together.

File: subworkflows/local/prepare_genome/main.nf [hunk 4]
```diff
@@ -55,7 +55,7 @@ workflow PREPARE_GENOME {
                 ch_bowtie_index = UNTAR_BOWTIE_INDEX.out.files
                 ch_versions  = ch_versions.mix(UNTAR_BOWTIE_INDEX.out.versions)
             } else {
-                ch_bowtie_index = Channel.fromPath("${val_bowtie_index}**ebwt", checkIfExists: true).map{it -> [ [id:it.baseName], it ] }.collect()
+                ch_bowtie_index = Channel.fromPath("${val_bowtie_index}**ebwt", checkIfExists: true).map{it -> [ [id:'bowtie_index'], it ] }.collect()
```
Comment (atrigila) [inline subworkflows/local/prepare_genome/main.nf:58]: `Mirdeep2` is a bit tricky in how the index should be named: https://github.com/nf-core/smrnaseq/blob/adc43dbbe5ce454939aeb7e3717b24d75b44e5c8/modules/nf-core/mirdeep2/mapper/main.nf#L30.

So if the path to the index `ch_bowtie_index` is something like:
`/work/bowtie_index`
and the files inside that directory are:
`/work/bowtie_index/genome.1.ebwt`
`/work/bowtie_index/genome.2.ebwt`
`/work/bowtie_index/genome.3.ebwt`

Then the meta should be: `genome`.

This renaming is done automatically when indexing the fasta in the else statement:


```
            // Index FASTA with nf-core Bowtie1
            INDEX_GENOME ( CLEAN_FASTA.out.output )
            ch_versions      = ch_versions.mix(INDEX_GENOME.out.versions)
```

File: subworkflows/local/prepare_genome/main.nf [hunk 5]
```diff
@@ -90,24 +90,24 @@ workflow PREPARE_GENOME {
...
-        ch_mirna_gtf         = params.mirgenedb_gff     ? Channel.fromPath(params.mirgenedb_gff, checkIfExists: true).collect()                                           : { exit 1, "MirGeneDB gff file not found via --mirgenedb_gff: ${params.mirgenedb_gff}"}
+        ch_mirna_gtf         = params.mirgenedb_gff     ? Channel.fromPath(params.mirgenedb_gff, checkIfExists: true).map{ it -> [ [id:it.baseName], it ] }.collect()     : { exit 1, "MirGeneDB gff file not found via --mirgenedb_gff: ${params.mirgenedb_gff}"}
     }
 
     emit:
     fasta                 = ch_fasta               // channel: [ val(meta), path(fasta) ]
     has_fasta             = bool_has_fasta         // boolean
-    bowtie_index          = ch_bowtie_index        // channel: [genome.1.ebwt, genome.2.ebwt, genome.3.ebwt, genome.4.ebwt, genome.rev.1.ebwt, genome.rev.2.ebwt]
+    bowtie_index          = ch_bowtie_index        // channel: [ val(meta), [ path(genome.1.ebwt), path(genome.2.ebwt), path(genome.3.ebwt), path(genome.4.ebwt), path(genome.rev.1.ebwt), path(genome.rev.2.ebwt) ] ]
```
Comment (atrigila) [inline subworkflows/local/prepare_genome/main.nf:99]: Same comment, due to how mirdeep2 is structured this should have the structure: `channel: [ val(meta), [ path(directory_index)] ]`

File: subworkflows/nf-core/fastq_find_mirna_mirdeep2/main.nf [hunk 6]
```diff
@@ -8,7 +8,7 @@ workflow FASTQ_FIND_MIRNA_MIRDEEP2 {
     take:
     ch_reads                     // channel: [ val(meta),  fastq  ]
     ch_genome_fasta              // channel: [ val(meta),  genome_fasta  ]
-    ch_bowtie_index              // channel: [ val(meta),  index  ]
+    ch_bowtie_index              // channel: [ val(meta), [ path(genome.1.ebwt), path(genome.2.ebwt), path(genome.3.ebwt), path(genome.4.ebwt), path(genome.rev.1.ebwt), path(genome.rev.2.ebwt) ] ]
```
Comment (atrigila) [inline subworkflows/nf-core/fastq_find_mirna_mirdeep2/main.nf:11]: This should not be changed in a nf-core subworkflow as it will break linting.

**Review by atrigila — COMMENTED** [review]
I've also run `nf-core lint` to check _TODOs left in the pipeline_ and found a few issues:

```
╭───────────────────────╮
│ LINT RESULTS SUMMARY  │
├───────────────────────┤
│ [✔] 460 Tests Passed  │
│ [?]   1 Test Ignored  │
│ [!]  26 Test Warnings │
│ [✗]   1 Test Failed   │
╰───────────────────────╯
```

File: subworkflows/local/genome_quant.nf (hunk 3 above)
Comment (nschcolnicov) [reply subworkflows/local/genome_quant.nf:10]: This is something that is a bit more tricky to update, I created a new ticket to address it: https://github.com/nf-core/smrnaseq/issues/451

File: subworkflows/local/prepare_genome/main.nf (hunk 4 above)
Comment (nschcolnicov) [reply subworkflows/local/prepare_genome/main.nf:58]: Also to be addressed in https://github.com/nf-core/smrnaseq/issues/451

File: subworkflows/local/prepare_genome/main.nf (hunk 5 above)
Comment (nschcolnicov) [reply subworkflows/local/prepare_genome/main.nf:99]: https://github.com/nf-core/smrnaseq/issues/451

File: subworkflows/nf-core/fastq_find_mirna_mirdeep2/main.nf (hunk 6 above)
Comment (nschcolnicov) [reply subworkflows/nf-core/fastq_find_mirna_mirdeep2/main.nf:11]: You are right, lets take care of it in https://github.com/nf-core/smrnaseq/issues/451

**Review by apeltzer — APPROVED** [review]

**Pull request merged**